File: src/cucumber/java/io/tripled/social/client/HelloWorldStepDefs.java
package io.tripled.social.client;

import cucumber.api.java.en.Given;
import cucumber.api.java.en.Then;
import cucumber.api.java.en.When;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;

public class HelloWorldStepDefs {

  private String myName;
  private String response;

  @Given("^the system knows my name is \"(.*)\"$")
  public void theSystemKnowsMyNameIs(String name) {
    this.myName = name;
  }

  @When("^I say hi to the system$")
  public void iSayHiToTheSystem() {
    response = "Hello, " + myName;
  }

  @Then("^the system responds with \"(.*)\"$")
  public void theSystemRespondsWith(String expectedResponse) {
    assertThat(response, equalTo(expectedResponse));
  }
}

File: src/test/kotlin/io/tripled/social/client/acceptance/HelloWorldStepDefs.kt
package io.tripled.social.client.acceptance

import io.cucumber.java.en.Given
import io.cucumber.java.en.Then
import io.cucumber.java.en.When
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers

class HelloWorldStepDefs {
    private var myName: String? = null
    private var response: String? = null

    @Given("^the system knows my name is \"(.*)\"$")
    fun theSystemKnowsMyNameIs(name: String?) {
        this.myName = name
    }

    @When("^I say hi to the system$")
    fun iSayHiToTheSystem() {
        response = "Hello, " + myName
    }

    @Then("^the system responds with \"(.*)\"$")
    fun theSystemRespondsWith(expectedResponse: String) {
        MatcherAssert.assertThat<String?>(response, Matchers.equalTo<String>(expectedResponse))
    }
}
